transform-plugins/src/main/java/co/cask/hydrator/plugin/Hasher.java

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.hydrator.plugin;
+
+import co.cask.cdap.api.annotation.Description;
+import co.cask.cdap.api.annotation.Name;
+import co.cask.cdap.api.annotation.Plugin;
+import co.cask.cdap.api.data.format.StructuredRecord;
+import co.cask.cdap.api.data.schema.Schema;
+import co.cask.cdap.api.plugin.PluginConfig;
+import co.cask.cdap.etl.api.Emitter;
+import co.cask.cdap.etl.api.PipelineConfigurer;
+import co.cask.cdap.etl.api.Transform;
+import co.cask.cdap.etl.api.TransformContext;
+import org.apache.commons.codec.digest.DigestUtils;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import javax.annotation.Nullable;
+
+/**
+ * Hash field(s) values using one the the digest algorithms.
+ */
+@Plugin(type = "transform")
+@Name("Hasher")
+@Description("Encodes field values using one of the digest algorithms. MD2, MD5, SHA1, SHA256, " +
+  "SHA384 and SHA512 are the supported message digest algorithms.")
+public final class Hasher extends Transform<StructuredRecord, StructuredRecord> {
+  private final Config config;
+  private Set<String> fieldsMap = new HashSet<>();
+
+
+  // For testing purpose only.
+  public Hasher(Config config) {
+    this.config = config;
+  }
+
+  @Override
+  public void initialize(TransformContext context) throws Exception {
+    super.initialize(context);
+    // Split the fields to be hashed.
+    String[] fields = config.fields.split(",");
+    for (String field : fields) {
+      fieldsMap.add(field);
+    }
+  }
+
+  @Override
+  public void configurePipeline(PipelineConfigurer pipelineConfigurer) throws IllegalArgumentException {
+    super.configurePipeline(pipelineConfigurer);
+    
+    // Checks if hash specified is one of the supported types. 
+    if (!config.hash.equalsIgnoreCase("md2") && !config.hash.equalsIgnoreCase("md5") &&
+        !config.hash.equalsIgnoreCase("sha1") && !config.hash.equalsIgnoreCase("sha256") &&
+        !config.hash.equalsIgnoreCase("sha384") && !config.hash.equalsIgnoreCase("sha512")) {
```

Comment:
I think the above condition would be easier to read and verify, if all supported digest algorithms are added into a set, and a `Set.contains(lowercase(specified digest))` is done. We could then use the same set for the exception message below.